### ISSUE TITLE
LICENSE Decisions between across yellr ecosystem as well as uploaded content

### ISSUE THREAD

**Issue by Nolski** [hhroc/yellr-server#80]
So I know we have discussed this a bit in the past but given that we will be presenting at Libre Planet I think we should pick a clear direction on how we want this software to be licensed. I am going to write up an email to licensing@fsf.org asking for advice and I'll CC @thequbit, as well as @decause (since it might be a good idea for Biz Legal of FOSS) as well as anyone else who is interested.

**Comment by thequbit** [comment]
Great idea!  Looking forward to seeing the draft email :).

Thanks for offering to do the leg work on this @Nolski!

**Comment by thequbit** [comment]
I wanted to circle back around and update this ticket with what @Nolski and I talked about on the phone.

the hhroc/yellr repo will disapear soon.  This repo is where we started off with the cordova app and the server, and the one-pager.  We'll be breaking all of that out into:

    hhroc/yellr-server
    hhroc/yellr-android
    hhroc/yellr-ios
    hhroc/yellr-one-pager ( this is up for discussion )

The license should definitely be the same across all of these ( I think it should at least, my hope is we'll get some insight from FSF ).

Thanks again for taking this on @Nolski !!!

**Comment by thequbit** [comment]
Additionally, we should pick what CC license we'll be going with for all uploaded content.  This is not quite as high priority, but as @Nolski has hinted at, it would be nice to have all of these buttoned up prior to Libre Planet.

**Comment by thequbit** [comment]
@Nolski did we get anything back from the SFF on this?

**Comment by Nolski** [comment]
Yup, So here is the response that I got

```
Hi Mike,

I'm sorry for the late response to this message.  Because we are a
nonprofit organization with very limited resources, messages to this
address often get backlogged, and we are always working hard to keep
up.

> We have had some discussion around using the AGPL, GPLv3, as well as the
> GPLv2 and have switched the license on our software a couple of times so
> far (with agreement from all of the developers). I wanted to get some
input
> from a lawyer who had experience in licensing free software to see which
> license may fit best for these two software platforms.  We also plan on
> applying a CC license to all user generated content submitted through the
> app. If there was someone who could touch on which CC license to use, that
> would help us as well.
>
> Do you have any advice or input?

Our recommendations are here:

  https://www.gnu.org/licenses/license-recommendations.html

In your particular case, since you are writing server software, we would
recommend the latest version of the AGPL. As for the CC license, our
recommendations point towards the latest version of CC BY-SA.

Please note however, that I'm not a lawyer (just a licensing volunteer
for the FSF) and that this is not legal advice.

If you want to discuss this any further, please feel free to reply and
I'll do my best to answer quickly.
```

I replied asking about licensing of the Yellr android app. I have done a bit of research around the GPLv2 vs the GPLv3 and it looks like the GPLv3 provides much more than simply licensing hardware. http://www.ifross.org/en/what-difference-between-gplv2-and-gplv3

I do think the the AGPL is a good choice for yellr-server. We can simply put a link in our footer to the Github repo and our requirements should be met (I asked him if there was anything else that we should be required to do just in case). 

I think CC BY-SA seems like a fine choice for content.

**Comment by thequbit** [comment]
Very cool.  Looks like AGPL was the right way to go after all :).

Should we be placing a header on the top of each source file, ya think?  Or just stating it in the github README.md file and including the LICENSE file be sufficient?

We need to work on the ToS for the Android app.  That is where the CC BY-SA stuff would go.

Thanks for doing all that @Nolski !!!

**Comment by Nolski** [comment]
I agree. I think that (if possible) having a little modal pop up prior to submitting your first post, saying how we will be licensing all content submitted would be nice.

As for the AGPL, I think including it in the README and LICENSE file is fine.

**Comment by thequbit** [comment]
Very good.  Ticket opened for yellr-android to inform the user of the content licensing.

https://github.com/hhroc/yellr-android/issues/152

**Comment by thequbit** [comment]
This can be closed with #177.
